src/System.Linq/src/System/Linq/Last.cs

```diff
@@ -174,36 +107,49 @@ public static TSource LastOrDefault<TSource>(this IEnumerable<TSource> source, F
             OrderedEnumerable<TSource> ordered = source as OrderedEnumerable<TSource>;
             if (ordered != null)
             {
-                return ordered.LastOrDefault(predicate);
+                return ordered.TryGetLast(predicate, out found);
             }
 
             IList<TSource> list = source as IList<TSource>;
             if (list != null)
             {
                 for (int i = list.Count - 1; i >= 0; --i)
                 {
-                    TSource element = list[i];
-                    if (predicate(element))
+                    TSource result = list[i];
+                    if (predicate(result))
```

Comment:
I copied and pasted this from `Last`, which has a different local var name from `LastOrDefault`. That's why GitHub is showing diffs here.